src/main/kotlin/lotto/dto/InputLottoMachineRequestDto.kt

```diff
@@ -0,0 +1,35 @@
+package lotto.dto
+
+import lotto.domin.LottoNumberSet
+
+data class InputLottoMachineRequestDto(
+    val payment: Int,
+    val manualLottoCount: Int,
+    val manualLotto: List<LottoNumberSet>,
+) {
+
+    companion object {
+
+        const val CANNOT_CONVERT_INT = "정수로 변환할 수 없습니다."
+        const val NEGATIVE_ERROR = "음수는 입력할 수 없습니다."
+
+        fun of(inputPayment: String, inputManualLottoCount: String, manualLotto: List<LottoNumberSet>): InputLottoMachineRequestDto {
+            val payment: Int = inputPayment.toIntOrNull() ?: throw IllegalArgumentException(CANNOT_CONVERT_INT)
```

Comment:
```suggestion
            val payment = inputPayment.toIntOrNull() ?: throw IllegalArgumentException(CANNOT_CONVERT_INT)
```
IDE에서 hint 기능을 제공해주기 때문에, 코틀린의 장점인 타입 추론을 활용해도 좋을 것 같아요. (이하 관련 내용 동일)